recipes/websocketpp/all/conanfile.py

```diff
@@ -1,35 +1,48 @@
 import os
 from conans import ConanFile, tools
 
+required_conan_version = ">=1.33.0"
+
 
 class WebsocketPPConan(ConanFile):
     name = "websocketpp"
     description = "Header only C++ library that implements RFC6455 The WebSocket Protocol"
-    topics = ("conan", "websocketpp", "websocket", "network", "web", "rfc6455", "header-only")
+    topics = ("websocketpp", "websocket", "network", "web", "rfc6455", "header-only")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/zaphoyd/websocketpp"
     license = "BSD-3-Clause"
     settings = "os", "arch", "compiler", "build_type"
     exports_sources = ["patches/*"]
-    options = {"asio": ["boost", "standalone"]}
-    default_options = {"asio": "boost"}
-    
+    options = {
+        "asio": ["boost", "standalone", False],
+        "with_openssl": [True, False],
+        "with_zlib": [True, False],
+    }
+    default_options = {
+        "asio": "boost",
+        "with_openssl": True,
+        "with_zlib": True,
+    }
+
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
     def requirements(self):
-        self.requires("openssl/1.1.1k")
-        self.requires("zlib/1.2.11")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1l")
```

Comment:
Well, it should come from boost or asio actually. openssl is optional in asio. websocketpp indirectly depends on openssl through asio.

Reply:
Yeah, I guess that is a general "problem" with header-only libraries, since their actual dependencies depend on what headers you include and what features you enable through defines. Personally I would prefer those dependencies to be left up to the user, and not included by default, but I am too new to conan to be able to see the full consequences of this suggestion.

But do you want me to do anything about this dependency in particular?

Reply:
Should I remove it completely? And does that go for the zlib dependency too?

Reply:
> But do you want me to do anything about this dependency in particular?

> Should I remove it completely?

I guess it's fine right now, because asio recipe doesn't require openssl (but it should).

> And does that go for the zlib dependency too?

It seems to be a direct dependency of websocketpp, so no.

Reply:
Ok! I'll leave it like it is then!